etp-backend/src/test/clj/solita/etp/service/energiatodistus_pdf_test.clj

```diff
@@ -40,17 +47,22 @@
   (t/is (= "12,346 %" (formats/format-number 0.1234567 3 true))))
 
 (t/deftest fill-xlsx-template-test
-  (doseq [energiatodistus (energiatodistukset)]
-    (let [path (service/fill-xlsx-template energiatodistus "fi" false)
-          file (-> path io/input-stream)
-          loaded-xlsx (xlsx/load-xlsx file)
-          sheet-0 (xlsx/get-sheet loaded-xlsx 0)]
+  (let [{:keys [energiatodistukset]} (test-data-set)
```

Comment:
I wonder how this has ever worked. It has compared numbers with strings and tried to find id in wrong cell.

Reply:
Just realized: since energiatodistukset where generated but not inserted to db in the previous version, the keys :id and :versio always had nil values. And because the cell was wrong, id nil was compared with nil cell => worked.